skimage/color/colorconv.py

```diff
@@ -192,20 +185,30 @@ def rgba2rgb(rgba, background=(1, 1, 1)):
     >>> img_rgba = data.logo()
     >>> img_rgb = color.rgba2rgb(img_rgba)
     """
-    arr = _prepare_rgba_array(rgba)
-    if isinstance(background, tuple) and len(background) != 3:
-        raise ValueError('the background must be a tuple with 3 items - the '
-                         'RGB color of the background. Got {0} items.'
-                         .format(len(background)))
-
-    alpha = arr[..., -1]
-    channels = arr[..., :-1]
-    out = np.empty_like(channels)
-
-    for ichan in range(channels.shape[-1]):
-        out[..., ichan] = np.clip(
-            (1 - alpha) * background[ichan] + alpha * channels[..., ichan],
-            a_min=0, a_max=1)
+    arr = np.asanyarray(rgba)
+
+    if arr.ndim < 2 or arr.shape[-1] != 4:
+        msg = ("the input array must have a shape == (.., 4)), "
+               "got {0}".format(arr.shape))
+        raise ValueError(msg)
+
+    arr = dtype.img_as_float(arr)
+
+    background = np.ravel(background).astype(arr.dtype)
+    if len(background) != 3:
+        raise ValueError('background must be an iterable containing 3 RGB '
+                         'values. Got {0} items'.format(len(background)))
+    if np.any(background < 0) or np.any(background > 1):
+        raise ValueError('background RGB values must be floats between '
+                         '0 and 1.')
+
+    background = background[np.newaxis, ...]
+    alpha = arr[..., -1, np.newaxis]
+    channels = arr[np.newaxis, ..., :-1]
```

Comment:
Broadcasting will work between `background` and `channels`, but unfortunately not between `alpha` and `background`:
```python
>>> from numpy.random import rand
>>> import numpy as np
>>> 
>>> background = rand(3)
>>> alpha = rand(5, 6)
>>> channels = rand(5, 6, 3)
>>> out = np.clip((1 - alpha) * background + alpha * channels,
...                a_min=0, a_max=1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: operands could not be broadcast together with shapes (5,6) (3,) 
```

Reply:
I meant removing it only in channels, not alpha

Reply:
OK, I see, sorry @jni... It works in fact, but the `np.squeeze` is still needed when the input is 1D (`rgba.shape == (4, )`), because in this case `out.shape ==  (1, 3)`:
```python
    background = background[np.newaxis, ...]
    alpha = arr[..., -1, np.newaxis]
    channels = arr[..., :-1]

    out = np.clip((1 - alpha) * background + alpha * channels,
                  a_min=0, a_max=1)

    if arr.ndim == 1:
        # To ensure that output ndim is the same input ndim
        out = np.squeeze(out, axis=0)
```
Do you think this modification is necessary?

Reply:
Let's forget about this! Thanks for the clarification! =)